app/src/main/java/fraktal/io/ext/Reducer.kt

```diff
@@ -26,38 +23,40 @@ fun <C, S, E> Reducer(
 
 class Reducer<C, S, U, E>(
     private val decider: Decider<C, S, E>,
-    private val scope: CoroutineScope,
-    private val uiMapper: (S) -> U,
-
-    private val _states: MutableStateFlow<S> = MutableStateFlow(decider.initialState),
-    private val _uiStates: MutableStateFlow<U> = MutableStateFlow(uiMapper(decider.initialState)),
-    private val commandChannel: Channel<C> = Channel<C>(capacity = CONFLATED),
-
-    private val _events: Channel<E?> = Channel(capacity = CONFLATED),
-) {
-    val uiStates = _uiStates.asStateFlow()
+    private val view: View<U, E>,
+    scope: CoroutineScope,
+    capacity: Int = CONFLATED
+    ) {
+    private val _deciderState: MutableStateFlow<S> = MutableStateFlow(decider.initialState)
+    private val _viewState: MutableStateFlow<U> = MutableStateFlow(view.initialState)
+    private val _commands: Channel<C> = Channel(capacity = capacity)
+    private val _events: Channel<E?> = Channel(capacity = capacity)
+
+    val uiStates = _viewState.asStateFlow()
     val events = _events.consumeAsFlow()
 
     init {
         scope.launch {
-            commandChannel.consumeAsFlow().collectLatest { command ->
+            // Only Collect LATEST is supported ???
```

Comment:
To be honest, this is a very controversial decision. related to whether we want to maintain a queue of commands or not. I can't answer unequivocally

Reply:
OK. I wanted to highlight this, as in some situations you might want to do this differently and not cancel the flow with `collectLatest`